src/endpoints/http/collections/create_test.js

```diff
@@ -1,61 +1,28 @@
-import supertest from 'supertest';
-import express from 'express';
-import bodyParser from 'body-parser';
-import {createCollection} from './create.js';
-
-const collectionEndPoint = '/collections';
-const app = express();
-app.use(bodyParser.json());
-app.use((req, res, next) => {
-  req.db = {
-    createCollection: (name, cb) => {
-      if (name === 'fail') {
-        return cb(new Error('db error'));
-      }
-
-      cb();
-    }
-  };
-
-  req.log = {
-    debug: function() {},
-    trace: function() {}
-  };
-
-  next();
-});
-
-export function testNameParamRequired(done) {
-  createCollection(app);
-
-  supertest(app)
-    .post(collectionEndPoint)
-    .expect(400)
-    .expect(res => {
-      res.message = 'name is required';
-    })
-    .end(done);
+import assert from 'assert';
+import createCollection from './create';
+import sinon from 'sinon';
+
+const mockLogger = {
+  debug: () => {}
+};
+
+const createCollectionStub = sinon.stub();
+
+const mockDb = {
+  createCollection: createCollectionStub
+};
+
+export function testCreateCollection(done) {
+  createCollectionStub.yields(null);
+  createCollection('test-create-collection', mockLogger, mockDb, 'testCollection').then(result => {
+    assert.equal(result, 'testCollection');
+  });
+  done();
```

Comment:
move the `done` function to the above callback. Otherwise the assert statements above won't be executed.